### PR TITLE
Update sdk to v2

### PR DESCRIPTION
**Pull request by roppazvan**
### Why does this PR exist?

This updates the studio sync provider to use the latest SDK version. 

### Testing this change

Run graphql locally. Add the local endpoint in the plugin env variables. 

### Additional Notes (if any)

Screenshot 2024-11-22 at 17 19 02

### CODE REVIEW

File: packages/tokens-studio-for-figma/src/utils/themeListToTree.ts [hunk 1]
```diff
@@ -41,15 +43,17 @@ export function themeListToTree(items: ThemeObjectsList) {
...
+      const childrenLength = acc.filter(
+        (item) => item.isLeaf === true
+          && ((typeof curr?.group !== 'undefined' && item.parent === curr.group)
+            || (typeof curr?.group === 'undefined' && item.parent === INTERNAL_THEMES_NO_GROUP)),
+      ).length;
       acc.splice(parentIndex + childrenLength + 1, 0, {
         isLeaf: true,
         value: curr,
-        key: curr.id,
+        key: curr.name,
```
Comment (six7) [inline packages/tokens-studio-for-figma/src/utils/themeListToTree.ts:54]: Is this a fn just used by Studio?

File: packages/tokens-studio-for-figma/src/utils/tokenHelpers.ts [hunk 2]
```diff
@@ -4,17 +4,20 @@ import { ThemeObject, UsedTokenSetsMap } from '@/types';
...
+true,
+{
+  failedToResolve?: boolean;
+}
+>;
 
 export function getOverallConfig(themes: ThemeObject[], selectedThemes: string[]) {
   return selectedThemes.reduce((acc, themeId) => {
-    const currentTheme = themes.find((theme) => theme.id === themeId);
+    const currentTheme = themes.find((theme) => theme.name === themeId);
```
Comment (six7) [inline packages/tokens-studio-for-figma/src/utils/tokenHelpers.ts:16]: Why are we switching from id to name?

**Review by six7 — COMMENTED** [review]
It'd have been easier if formatting changes were not part of that PR. Also, it had gotten super big, it might have made sense to break it down in smaller PRs.

File: packages/tokens-studio-for-figma/src/app/store/models/reducers/tokenState/renameVariableNamesToThemes.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ export function renameVariableNamesToThemes(state: TokenState, tokensToRename: T
     return acc;
   }, {});
   const newThemes = state.themes.map((theme) => {
-    const updatedTokens = theme.$figmaVariableReferences;
+    const updatedTokens = { ...theme.$figmaVariableReferences };
```
[the remark this answers is not in the document]
Comment (roppazvan) [reply packages/tokens-studio-for-figma/src/app/store/models/reducers/tokenState/renameVariableNamesToThemes.ts]: reverted this change as it was causing issues with tests.

File: packages/tokens-studio-for-figma/src/app/store/utils/updateAliasesInState.test.ts [hunk 4]
```diff
@@ -84,8 +84,8 @@ describe('updateAliasesInState', () => {
       ],
     };
 
-    const updatedTokens = updateAliasesInState(tokens, data);
+    const { newTokens } = updateAliasesInState(tokens, data);
 
-    expect(updatedTokens).toEqual(expectedTokens);
+    expect(newTokens).toEqual(expectedTokens);
```
[the remark this answers is not in the document]
Comment (roppazvan) [reply packages/tokens-studio-for-figma/src/app/store/utils/updateAliasesInState.test.ts]: just renamed it, but reverted back. updatedTokens is better.

File: packages/tokens-studio-for-figma/src/utils/themeListToTree.ts (hunk 1 above)
Comment (roppazvan) [reply packages/tokens-studio-for-figma/src/utils/themeListToTree.ts:54]: it's used by the manage themes modal

File: packages/tokens-studio-for-figma/src/utils/tokenHelpers.ts (hunk 2 above)
Comment (roppazvan) [reply packages/tokens-studio-for-figma/src/utils/tokenHelpers.ts:16]: this was done before we moved everything to postgres . The idea was that gitea would be name based. Most of the studio stuff is the same. everything is based on name, except project and org.

Comment (six7) [reply packages/tokens-studio-for-figma/src/utils/tokenHelpers.ts:16]: @roppazvan i think we cannot compare name.. there could be multiple themes with the same name (under different groups), lets keep to id

File: packages/tokens-studio-for-figma/src/utils/themeListToTree.ts (hunk 1 above)
Comment (six7) [reply packages/tokens-studio-for-figma/src/utils/themeListToTree.ts:54]: i think same as before, name is not unique

**Review by six7 — APPROVED** [review]
approving code-side, @rbosker @keeganedwin can you do functional testing?

**Comment by keeganedwin** [comment]
> approving code-side, @rbosker @keeganedwin can you do functional testing?
Test was successful. Was able to :
- Choose Org
- Choose Proj
- Export styles & variables

**Pull request merged**